File: HelloTuiLua/src/recombine/recombineui.lua
require "tagMap/Tag_recombineui"

Recombineui = class("Recombineui",function()
	return TuiBase:create()
end)

Recombineui.__index = Recombineui
local __instance = nil

function Recombineui:create()
	local ret = Recombineui.new()
	__instance = ret
	ret:setOnLoadSceneScriptHandler(function() ret:onLoadScene() end)
	return ret
end

function Recombineui:getControl(tagPanel,tagControl)
	local ret = nil
	ret = self:getPanel(tagPanel):getChildByTag(tagControl)
	return ret
end

function Recombineui:getPanel(tagPanel)
	local ret = nil
	if tagPanel == Tag_recombineui.PANEL_RECOMBINE then
		ret = self:getChildByTag(tagPanel)
	end
	return ret
end
---------------logic----------------------------
local function event_btn_back(p_sender)
	CSceneManager:getInstance():replaceScene(
		CCSceneExTransitionFade:create(0.5,LoadScene("Welcomeui")))
end

local function event_btn_click(p_sender)
	print("tag : "..tolua.cast(p_sender,"ccw.CWidget"):getUserTag())
end

local function event_adapt_gpvbag(p_convertview, idx)
	local pCell = p_convertview
	local btn = nil
	if pCell == nil then
		pCell = CGridPageViewCell:new()

		TuiManager:getInstance():parseCell(pCell, "cell_grid", PATH_RECOMBINEUI)
		btn = pCell:getChildByTag(Tag_recombineui.BTN_GRID)
		btn:setOnClickScriptHandler(event_btn_click)
	else
		btn = pCell:getChildByTag(Tag_recombineui.BTN_GRID)
	end
	tolua.cast(btn,"ccw.CWidget"):setUserTag(idx)
	return pCell
end

function Recombineui:onLoadScene()
	-- cc.SpriteFrameCache:getInstance():addSpriteFrames("recombine/recombineui.plist")
	TuiManager:getInstance():parseScene(self,"panel_recombine",PATH_RECOMBINEUI)

	local btnBack = self:getControl(Tag_recombineui.PANEL_RECOMBINE,Tag_recombineui.BTN_BACK)
	btnBack:setOnClickScriptHandler(event_btn_back)
	
	local gpvBag = self:getControl(Tag_recombineui.PANEL_RECOMBINE,Tag_recombineui.GPV_BAG)
	gpvBag:setDataSourceAdapterScriptHandler(event_adapt_gpvbag)
	gpvBag:reloadData()
	
end

File: HelloTuiCpp/Classes/scene/Recombine/Recombineui.cpp
#include "Recombineui.h"
#include "tui/tagMap/tuiTag_recombineui.h"
#include "tui/TuiManager.h"

void Recomebineui::onLoadScene()
{
	TuiManager::getInstance()->parseScene(this, "panel_recombine", PATH_RECOMBINEUI);

	CGridPageView *pGpvBag = (CGridPageView*)this->getControl(PANEL_RECOMBINE, GPV_BAG);
	pGpvBag->setDataSourceAdapter(this,ccw_datasource_adapter_selector(Recomebineui::adapt_gpv_bag));
	pGpvBag->reloadData();

	CButton *pBtnBack = (CButton*)this->getControl(PANEL_RECOMBINE, BTN_BACK);
	pBtnBack->setOnClickListener(this, ccw_click_selector(Recomebineui::event_btn_back));
}

Ref* Recomebineui::adapt_gpv_bag(Ref* pConvertCell, unsigned int uIdx)
{
	CGridPageViewCell* pCell = (CGridPageViewCell*)pConvertCell;
	CButton *btn = nullptr;
	if (!pCell)
	{
		pCell = new CGridPageViewCell();
		pCell->autorelease();

		TuiManager::getInstance()->parseCell(pCell, "cell_grid", PATH_RECOMBINEUI);
		btn = (CButton*)pCell->getChildByTag(BTN_GRID);
		btn->setOnClickListener(this, ccw_click_selector(Recomebineui::event_btn_click));
	}
	else{
		btn = (CButton*)pCell->getChildByTag(BTN_GRID);
	}
	btn->setUserTag(uIdx);
	return pCell;
}

void Recomebineui::event_btn_click(Ref* pSender)
{
	CButton *btn = (CButton*)pSender;
	int id = btn->getUserTag();

	CCLOG("id = %d", id);
}

void Recomebineui::event_btn_back(Ref* pSender)
{
	CSceneManager::getInstance()->replaceScene(
		CCSceneExTransitionFade::create(0.5f, LoadScene("Main::Mainui")));
}

/************************************************************************/
//	GET/SET/IS
/************************************************************************/

Node * Recomebineui::getPanel(int tagPanel)
{
	Node *pPanel = nullptr;
	switch (tagPanel)
	{
	case PANEL_RECOMBINE:
		pPanel = this->getChildByTag(PANEL_RECOMBINE);
		break;
	}
	return pPanel;
}
